tests/ts/examples.ts

```diff
@@ -284,9 +284,23 @@ qb = qb.where(raw('random()', 1, '2'));
 qb = qb.where(Person.raw('random()', 1, '2', raw('3')));
 qb = qb.alias('someAlias');
 
-// query builder hooks
-qb = qb.runBefore(async (result: any, builder: objection.QueryBuilder<Person>) => {});
-qb = qb.runAfter(async (result: Person[], builder: objection.QueryBuilder<Person>) => {});
+// Query builder hooks. runBefore() and runAfter() don't immediately affect the result.
+
+const runBeforePerson: Promise<Person> = qb
+  .first()
+  .throwIfNotFound()
+  .runBefore(async (result: any, builder: objection.QueryBuilder<Person>) => 88);
+const runBeforePersons: Promise<Person[]> = qb.runBefore(
+  async (result: any, builder: objection.QueryBuilder<Person, Person[]>) => 88
+);
+
+const runAfterPerson: Promise<Person> = qb
+  .first()
+  .throwIfNotFound()
+  .runAfter(async (result: any, builder: objection.QueryBuilder<Person>) => 88);
+const runAfterPersons: Promise<Person[]> = qb.runAfter(
+  async (result: any, builder: objection.QueryBuilder<Person, Person[]>) => 88
+);
 
```

Comment:
@koskimas, perhaps you can review these runBefore/runAfter tests? I'm mainly ensuring that these methods **do not** affect the query builder type chain, at least not in-place.